components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/threatprotection/analyzer/JSONAnalyzer.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.threatprotection.analyzer;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import org.wso2.carbon.apimgt.gateway.threatprotection.APIMThreatAnalyzerException;
+import org.wso2.carbon.apimgt.gateway.threatprotection.configuration.JSONConfig;
+import org.wso2.carbon.apimgt.gateway.threatprotection.configuration.XMLConfig;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * Implementation of APIMThreatAnalyzer for JSON Payloads
+ */
+public class JSONAnalyzer implements APIMThreatAnalyzer {
+
+    private static final String JSON_THREAT_PROTECTION_MSG_PREFIX = "Threat Protection-JSON: ";
+    private JsonFactory factory;
+    private boolean enabled = true;
+    private int maxFieldCount = 0;
+    private int maxStringLength = 0;
+    private int maxArrayElementCount = 0;
+    private int maxFieldLength = 0;
+    private int maxJsonDepth = 0;
+
+    public JSONAnalyzer() {
+        factory = new JsonFactory();
+    }
+
+    /**
+     * Create a JSONAnalyzer using default configuration values
+     */
+    public void configure(JSONConfig config) {
+        enabled = config.isEnabled();
+        maxFieldCount = config.getMaxPropertyCount();
+        maxStringLength = config.getMaxStringLength();
+        maxArrayElementCount = config.getMaxArrayElementCount();
+        maxFieldLength = config.getMaxKeyLength();
+        maxJsonDepth = config.getMaxJsonDepth();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public void configure(XMLConfig config) {
+        throw new UnsupportedOperationException("This method is not supported on this instance");
+    }
+
+    /**
+     * @param in payload
+     * @throws APIMThreatAnalyzerException if defined limits for json payload exceeds
+     */
+    @Override
```

Comment:
Incomplete Java doc comment. Since this is a overidden method, we do not need to have a method comment, if there is anything specific to this implementation